src/plugins/code-block.tsx

```diff
@@ -65,61 +67,82 @@ const highlight = (code, language) => {
   }
 
   const html = `<div slot="content" class="highlight highlight-source${
-    languageIsSet ? `-${language}` : ""
+    languageIsSet ? `-${language}` : ''
   }">${highlighted}</div>`;
 
-  const lineCount = html.split(/\r\n|\r|\n/).length;
+  let lineCount = html.split(/\r\n|\r|\n/).length;
+  let regionalCopy = false;
+
+  if (regions > 1 && idx > 0) {
+    lineCountOffSet = lineCountOffSet + lineCount;
+    regionalCopy = true;
+  } else {
+    lineCountOffSet = 0;
+  }
 
   return [
     {
-      type: "element",
-      tagName: "p",
+      type: 'element',
+      tagName: 'p',
       properties: {
-        class: "searchable-code",
+        class: 'searchable-code'
       },
       children: [
         {
-          type: "text",
-          value: `${entities.encode(code)}`,
-        },
-      ],
+          type: 'text',
+          value: `${entities.encode(code)}`
+        }
+      ]
     },
     {
-      type: "jsx",
-      value: `<CodeBlock language="${language}" lineCount="${lineCount}">`,
+      type: 'jsx',
+      value: `<CodeBlock language="${language}" lineCount="${lineCount}" lineCountOffset="${lineCountOffSet}" regionalCopy="${regionalCopy}" >`
     },
     ...unified()
-      .use(rehypeParse, {fragment: true})
+      .use(rehypeParse, { fragment: true })
       .parse(html).children,
     {
-      type: "jsx",
-      value: `</CodeBlock>`,
-    },
+      type: 'jsx',
+      value: `</CodeBlock>`
+    }
   ];
 };
 
 const addVersions = (code) => {
   code = code.replace(/ANDROID_VERSION/g, versions.ANDROID_VERSION);
   code = code.replace(
     /ANDROID_KOTLIN_VERSION/g,
-    versions.ANDROID_KOTLIN_VERSION,
+    versions.ANDROID_KOTLIN_VERSION
   );
   code = code.replace(/ANDROID_GEO_VERSION/g, versions.ANDROID_GEO_VERSION);
+
+  return code;
+};
+
+const removeCopy = (code) => {
+  code = code.split('###BEGIN_COPY###');
   return code;
 };
 
+const reducer = (previousValue, currentValue) =>
```

Comment:
personally I think I'd define this within the `visit` function, just because that's the only place it's used.  up to you though!